crates/levm/src/vm.rs

```diff
@@ -452,14 +471,105 @@ impl VM {
                         .memory
                         .copy(src_offset, dest_offset, size);
                 }
+                Opcode::CALL => {
+                    let gas = current_call_frame.stack.pop().unwrap();
+                    let address =
+                        Address::from_low_u64_be(current_call_frame.stack.pop().unwrap().low_u64());
+                    let value = current_call_frame.stack.pop().unwrap();
+                    let args_offset = current_call_frame.stack.pop().unwrap().try_into().unwrap();
+                    let args_size = current_call_frame.stack.pop().unwrap().try_into().unwrap();
+                    let ret_offset = current_call_frame.stack.pop().unwrap().try_into().unwrap();
+                    let ret_size = current_call_frame.stack.pop().unwrap().try_into().unwrap();
+
+                    // check balance
+                    if self.balance(&current_call_frame.msg_sender) < value {
+                        current_call_frame.stack.push(U256::zero());
+                        continue;
+                    }
+
+                    // transfer value
+                    // transfer(&current_call_frame.msg_sender, &address, value);
+
+                    let callee_bytecode = self.get_account_bytecode(&address);
+
+                    if callee_bytecode.is_empty() {
+                        current_call_frame.stack.push(U256::one());
+                        continue;
+                    }
+
+                    let calldata = current_call_frame
+                        .memory
+                        .load_range(args_offset, args_size)
+                        .into();
+
+                    let new_call_frame = CallFrame {
+                        gas,
+                        msg_sender: current_call_frame.msg_sender, // caller remains the msg_sender
+                        callee: address,
+                        bytecode: callee_bytecode,
+                        msg_value: value,
+                        calldata,
+                        ..Default::default()
+                    };
+
+                    current_call_frame.return_data_offset = Some(ret_offset);
+                    current_call_frame.return_data_size = Some(ret_size);
+
+                    self.call_frames.push(current_call_frame.clone());
```

Comment:
Shouldn't this be the `new_call_frame`?

Reply:
Ah nevermind, we are pushing the `current_call_frame` so we have it as parent and we change the `current_call_frame` later

Reply:
It's kind of confusing, but no, because we store in the `callframes` vec the callframes that are pending to execute, in other words the ones that give control of the execution to a sub call, which then needs to return back to their parent. So the parents are held in the vec, and the current callframe continues to be new one.